Source/WebCore/rendering/RenderObject.cpp

```diff
@@ -2650,10 +2673,21 @@ auto RenderObject::collectSelectionGeometriesInternal(const SimpleRange& range)
     bool containsDifferentWritingModes = false;
     bool hasLeftToRightText = false;
     bool hasRightToLeftText = false;
+    bool needsToSeparateBoundary = false;
+    RenderObject* lastRenderer = nullptr;
```

Comment:
You should probably use `WeakPtr` here instead of a raw pointer.

(`CheckedPtr` may not be viable here because there's a potential call to `VisiblePosition::VisiblePosition` below, which may update layout and therefore destroy the renderer that `lastRenderer` points to).

Reply:
Done, thanks:)